sky/server/requests/executor.py

```diff
@@ -408,9 +409,20 @@ def start(deploy: bool) -> List[multiprocessing.Process]:
         queue_names = [
             schedule_type.value for schedule_type in api_requests.ScheduleType
         ]
+        # TODO(aylei): queue manager port should be configurable.
         queue_server = multiprocessing.Process(
             target=mp_queue.start_queue_manager, args=(queue_names,))
         queue_server.start()
+
+        # Monitor the queue server process and raise an error if it exits to
+        # avoid the main process hanging on the queue server.
```

Comment:
Hmm, I am a bit confused by this comment. IIRC, exception raised in a thread will not affect the main thread. How could this monitor help with the hanging then?

Reply:
Good catch! I mistook the exception would pop up to main process😅 Given this, I think explicitly checking the port available works best for now.